tests/models/test_responses.py

```diff
@@ -298,6 +298,22 @@ def test_response_force_encoding():
     assert response.encoding == "iso-8859-1"
 
 
+def test_response_force_encoding_after_text_accessed():
+    response = httpx.Response(
+        200,
+        content=b"Hello, world!",
+    )
+    assert response.status_code == 200
+    assert response.reason_phrase == "OK"
+    assert response.text == "Hello, world!"
+    assert response.encoding == "utf-8"
+
+    response.encoding = "UTF8"
```

Comment:
I'd expect the `ValueError` to be raised here. I don't think we need the conditional "don't raise if the encoding is being set but the resulting codec won't change".

Simplicity over complexity where possible.